server/bin/pbench-backup-tarballs.py

```diff
@@ -0,0 +1,485 @@
+            quaran += 1
+            logger.error(
+                "Quarantine: {}, {} does not exist or it is not a regular file\n".format(tb, archive_md5))
+            continue
+
+        # read the md5sum from md5 file
+        try:
+            with open(archive_md5) as f:
+                archive_md5_hex_value = f.readline().split(" ")[0]
+        except (OSError, IOError) as e:
+            # Could not read file.
+            quarantine(qdir, logger, tb)
+            quaran += 1
+            logger.error(
+                "Quarantine: {}, Could not read {}, {}\n".format(tb, archive_md5, e))
+            continue
+
+        # match md5sum of the tarball to its md5 file
+        try:
+            with open(tar, 'rb') as f:
+                adata = f.read()
+                try:
+                    archive_tar_hex_value = hashlib.md5(adata).hexdigest()
+                except Exception as e:
+                    quarantine(qdir, logger, tb)
+                    quaran += 1
+                    logger.error(
+                        "Quarantine: {}, Unable to calculate the hexadecimal md5 value for {}, {}\n".format(tb, tar, e))
+                    continue
+        except (OSError, IOError) as e:
+            # Could not read file.
+            quarantine(qdir, logger, tb)
+            quaran += 1
+            logger.error(
+                "Quarantine: {}, Could not read {}, {}\n".format(tb, tar, e))
+            continue
+        else:
+            if archive_tar_hex_value == archive_md5_hex_value:
+                pass
+            else:
+                quarantine(qdir, logger, tb)
+                quaran += 1
+                logger.error(
+                    "Quarantine: {}, md5sum of {} does not match with its md5 file {}\n".format(tb, tar, archive_md5))
+                continue
+
+        resultname = os.path.basename(tar)
+        controller_path = os.path.dirname(tar)
+        controller = os.path.basename(controller_path)
+        backup_controller_path = "{}/{}".format(config.BACKUP, controller)
+
+        # This function call will handle all the local backup related
+        # operations and count the number of success and failure.
+        local_backup_result = backup_to_local(
+            config, logger, controller_path, controller, tb, tar, resultname, archive_md5, archive_md5_hex_value)
+        fnbackup_success += local_backup_result.nbackup_success
+        fnbackup_fail += local_backup_result.nbackup_fail
+
+        # This function call will handle all the S3 bucket related
+        # operations and count the number of success and failure.
+        s3_backup_result = backup_to_s3(
+            s3_obj, logger, controller_path, controller, tb, tar, resultname, archive_md5, archive_md5_hex_value)
+        fns3_success += s3_backup_result.ns3_success
+        fns3_fail += s3_backup_result.ns3_fail
+
+        if local_backup_result.status.name == "SUCCESS" and s3_backup_result.status.name == "SUCCESS":
```

Comment:
This should be `local_backup_result.status = Status.SUCCESS ...` - .name is not needed and no string comparisons